util/dockerutil/docker.go

```diff
@@ -0,0 +1,87 @@
+package dockerutil
+
+import (
+	"github.com/docker/buildx/store/storeutil"
+	"github.com/docker/cli/cli/command"
+	"github.com/docker/cli/cli/context/docker"
+	"github.com/docker/cli/opts"
+	"github.com/docker/docker/client"
+	"github.com/pkg/errors"
+)
+
+// Client represents an active docker object
+type Client struct {
+	CLI command.Cli
+}
+
+// NewDockerClient initializes a new docker client
+func NewDockerClient(cli command.Cli) *Client {
+	return &Client{CLI: cli}
+}
+
+// API returns a new Docker API client
+func (c *Client) API(name string) (client.APIClient, error) {
+	if name == "" {
+		name = c.CLI.CurrentContext()
+	}
+	return ClientForEndpoint(c.CLI, name)
+}
+
+// ValidateEndpoint validates that endpoint is either a context or a docker host
+func ValidateEndpoint(dockerCli command.Cli, ep string) (string, error) {
+	de, err := storeutil.GetDockerEndpoint(dockerCli, ep)
+	if err == nil && de != "" {
+		if ep == "default" {
+			return de, nil
+		}
+		return ep, nil
+	}
+
+	h, err := opts.ParseHost(true, ep)
+	if err != nil {
+		return "", errors.Wrapf(err, "failed to parse endpoint %s", ep)
+	}
+	return h, nil
+}
+
+// ClientForEndpoint returns a docker client for an endpoint
+func ClientForEndpoint(dockerCli command.Cli, name string) (client.APIClient, error) {
```

Comment:
I think this fits fine in storeutil as well(if there is any reason for them to be public at all). The imports are same.